tapchannel/aux_invoice_manager.go

```diff
@@ -198,6 +198,23 @@ func (s *AuxInvoiceManager) handleInvoiceAccept(_ context.Context,
 	// Convert the total asset amount to milli-satoshis using the price from
 	// the accepted quote.
 	rfqID := htlc.RfqID.ValOpt().UnsafeFromSome()
+
+	assetID, err := s.assetIDFromQuote(rfqID)
+	if err != nil {
+		return nil, fmt.Errorf("could not extract assetID from "+
+			"quote: %v", err)
+	}
+	for _, v := range htlc.Balances() {
+		if v.AssetID.Val != *assetID {
```

Comment:
This probably works for now as we only have a single asset ID in an HTLC. But this will definitely change with group key support. So perhaps we can add minimum forward compatibility by extracting this into a function, something like `htlcMatchesIdentifier(i asset.identifier)`. Then we can just change the logic there.